docs/DeveloperGuide.md

```diff
@@ -420,32 +421,148 @@ testers are expected to do more *exploratory* testing.
+   2. Test case: `unmark 1`<br>
+      Expected: Task "Bar" is marked as uncompleted. Details of the uncompleted task shown in the status message.
+   3. Test case: `unmark 1`<br>
+      Expected: No task is marked as uncompleted. Error details shown in the status message.
+   4. Test case: `unmark 0`<br>
+      Expected: No task is marked as uncompleted. Error details shown in the status message.
 
-### Deleting a person / task
+### Switching Lists
 
-1. Deleting a person / task while all persons / task are being shown
+1. List all tasks
 
-   1. Prerequisites: List all persons / task using the `list_contact` or `list_task` command.
+   1. Test case: `list_task`<br>
+      Expected: All tasks are listed in the task list and, if not already on the "Task List" view it's swapped to it. Details of the listed tasks shown in the status message.
+   2. Test case: `list_contact`<br>
+      Expected: All contacts are listed in the contact list and, if not already on the "Contact List" view it's swapped to it. Details of the listed contacts shown in the status message.
+   3. Test case: type `ctrl + tab`<br>
+      Expected: The current list is switched to the other list.
 
-   2. Test case: `delete_contact 1` / `delete_task 1`<br>
-      Expected: First contact / task is deleted from the list. Details of the deleted contact / task shown in the status message. Timestamp in the status bar is updated.
+### Viewing Task, Deadline and Contact Assocation
 
-   3. Test case: `delete_contact 0` / `delete_task 0`<br>
-      Expected: No person / task is deleted. Error details shown in the status message. Status bar remains the same.
+1. Select Task or Deadline to view Contact Assocation
 
+   1. Prerequisites: task "Foo" is already added and assigned to contact "Alex". If not added, add it first. Foo should be the first index in the task list.
+   2. Test case: `select_task 1`<br>
+      Expected: Task "Foo" is selected, the view is swapped to the "Task List" and the contact list is updated to show all contacts associated with the task. Details of the selected task shown in the status message.
+   3. Test case: `select_task 0`<br>
+      Expected: No task is selected. Error details shown in the status message.
+
+2. Select Contact to view Task or Deadline Association
+
+   1. Prerequisites: contact "Alex" is already added and assigned to task "Foo". If not added, add it first. Alex should be the first index in the contact list.
+   2. Test case: `select_contact 1`<br>
+      Expected: Contact "Alex" is selected, the view is swapped to the "Contact List" and the task list is updated to show all tasks associated with the contact. Details of the selected contact shown in the status message.
+   3. Test case: `select_contact 0`<br>
+      Expected: No contact is selected. Error details shown in the status message.
+
+### Finding Task, Deadline or Contact
```

Comment:
You might want to avoid using `Deadline` over here since we mainly only deal with two models, `Task` and `Contact`.

Reply:
Same for all other cases